java/compat/src/main/java/com/microsoft/bond/compat/CompatDriver.java

```diff
@@ -58,10 +58,9 @@ public static void main(final String[] args) throws IOException {
             System.exit(STATUS_FAILURE);
         }
 
-        final Compat compat = new Compat();
         final Serializer<Compat> serializer = new Serializer<>();
-        final Deserializer<Compat> deserializer = new Deserializer<>((StructBondType<Compat>) compat.getBondType());
-        deserializer.deserialize(reader);
+        final Deserializer<Compat> deserializer = new Deserializer<>((StructBondType<Compat>) Compat.BOND_TYPE);
```

Comment:
Please look at the original code. You initially created a new Compat instance and used its getBondType() to get the type descriptor, but the the result of deserialize() call was ignored. Thus, the Compat instance remained empty and the serialize() call produced empty result.

---
In reply to: [130488530](https://github.com/Microsoft/bond/pull/540#discussion_r130488530) [](ancestors = 130488530)